build/check-images.js

```diff
@@ -0,0 +1,164 @@
+// XXX Consider moving this to 'flaws.js'
+// Or, something. Checking for flaws should be very different from checking
+// for images.
+
+const path = require("path");
+
+const { Document, Image } = require("content");
+const { FLAW_LEVELS } = require("./constants");
+const { findMatchesInText } = require("./matches-in-text");
+
+/**
+ * Mutate the `$` instance for image reference and if appropriate,
+ * log them as flaws if they're not passing linting.
+ *
+ */
+function checkImageReferences(doc, $, options, { url, rawContent }) {
+  const filePaths = new Set();
+  if (doc.isArchive) return filePaths;
+
+  const checkImages = options.flawLevels.get("images") !== FLAW_LEVELS.IGNORE;
+
+  function addImageFlaw(src, id, { explanation, suggestion = null }) {
+    for (const match of findMatchesInText(src, rawContent, {
+      attribute: "src",
+    })) {
+      if (!("images" in doc.flaws)) {
+        doc.flaws.images = [];
+      }
+      doc.flaws.images.push(
+        Object.assign({ id, src, suggestion, explanation }, match)
+      );
+    }
+  }
+
+  const checked = new Set();
+  $("img[src]").each((i, element) => {
+    const img = $(element);
+    const src = img.attr("src");
+
+    // If the HTML contains the exact same 'src' value more than once,
+    // it will be found exactly that many times within the addImageFlaw()
+    // function which uses `findMatchesInText` so if it's repeated,
+    // it'll be logged for each individual occurance.
+    if (checked.has(src)) return;
+    checked.add(src);
+
+    // These two lines is to simulate what a browser would do.
+    const baseURL = `http://yari.placeholder${url}/`;
+    const absoluteURL = new URL(src, baseURL);
+
+    const imageID = `images${i}`;
```

Comment:
Not really an ID. But I'm not sure it matters. 
Suppose you have

```html
1| <img src="WrongCase.png">
2| and
3| <img src="WrongCase.png">
```
What you get in `index.json`'s `doc.flaws` is two entries and the final HTML looks like this:

```html
1| <img src="/en-US/docs/Web/HTML/WrongCase.png" data-flaw="image0">
2| and
3| <img src="/en-US/docs/Web/HTML/WrongCase.png" data-flaw="image0">
```
and in the `doc.flaws[]` you get the same message but with different `line` number.